### PR TITLE
Projectile improvements

### PR DESCRIPTION
**Pull request by TheEmidee**

### CODE REVIEW

File: Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp [hunk 1]
```diff
@@ -147,28 +147,28 @@ bool AGASExtProjectile::ShouldIgnoreHit_Implementation( AActor * other_actor, UP
     return instigator == nullptr || bIgnoreImpactWithInstigator && other_actor == instigator;
 }
 
-void AGASExtProjectile::PostProcessHit_Implementation( const FHitResult & /*hit_result*/ )
+void AGASExtProjectile::ApplyGameplayEffects()
 {
-    if ( !bShouldApplyGameplayEffectsOnDestroyed )
+    for ( const auto effect_spec : GameplayEffectContainerSpec.TargetGameplayEffectSpecHandles )
```
Comment (TheEmidee) [inline Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp]: const &

File: Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp [hunk 2]
```diff
@@ -147,28 +147,28 @@ bool AGASExtProjectile::ShouldIgnoreHit_Implementation( AActor * other_actor, UP
...
 }
 
-void AGASExtProjectile::PostProcessHit_Implementation( const FHitResult & /*hit_result*/ )
+void AGASExtProjectile::ApplyGameplayEffects()
 {
-    if ( !bShouldApplyGameplayEffectsOnDestroyed )
+    for ( const auto effect_spec : GameplayEffectContainerSpec.TargetGameplayEffectSpecHandles )
     {
-        ApplyGameplayEffects();
+        effect_spec.Data->GetContext().Get()->SetEffectCauser( this );
```
Comment (TheEmidee) [inline Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp]: cache `effect_spec.Data->GetContext().Get(` in a local variable

File: Source/GASExtensions/Public/Projectiles/GASExtProjectile.h [hunk 3]
```diff
@@ -53,21 +53,22 @@ class GASEXTENSIONS_API AGASExtProjectile : public AActor
     UFUNCTION( BlueprintImplementableEvent, meta = ( AllowPrivateAccess = true ) )
     void UpdateDestroyedGameplayCueParameters( FGameplayCueParameters & gameplay_cue_parameters );
 
+    // Called on both Server and Clients
     UFUNCTION( BlueprintImplementableEvent, meta = ( AllowPrivateAccess = true ) )
     void ReceiveOnHit( const FHitResult & hit_result );
 
     UFUNCTION( BlueprintNativeEvent, Category = "Projectile" )
```
Comment (TheEmidee) [inline Source/GASExtensions/Public/Projectiles/GASExtProjectile.h:60]: should be removed

File: Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp (hunk 1 above)
Comment (Frickerson) [reply Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp]: Done.

File: Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp (hunk 2 above)
Comment (Frickerson) [reply Source/GASExtensions/Private/Projectiles/GASExtProjectile.cpp]: Done.

File: Source/GASExtensions/Public/Projectiles/GASExtProjectile.h (hunk 3 above)
Comment (Frickerson) [reply Source/GASExtensions/Public/Projectiles/GASExtProjectile.h:60]: Done.

**Pull request merged**